craft_application/application.py

```diff
@@ -94,6 +94,10 @@ class Application:
     :ivar services: A ServiceFactory for this application
     """
 
+    # Feature flag for build-time secrets. Applications that want to use the feature
+    # must set this to True.
+    enable_build_secrets = False
```

Comment:
This seems like something that's more appropriate for `AppMetadata` since it's app configuration.

Reply:
oh that's a good point, I like this idea
do you have any thoughts on the "format" of this in `AppMetadata`? Is a "loose boolean" like this fine?

Reply:
I would do `features` and in there `build_secrets`

Reply:
I still think it's neat!